tasks/ThrusterTask.cpp

```diff
@@ -48,30 +48,35 @@ void ThrusterTask::updateHook()
     ThrusterTaskBase::updateHook();
 
     // Read Rock input port and update the message
-    if (_thrusters_cmd.readNewest( jointsCMD ) == RTT::NewData)
+    if (_thrusters_cmd.read( jointsCMD ) != RTT::NewData)
+        return;
+
+    if (jointsCMD.names.empty()) {
+        throw std::runtime_error("rock_gazebo::ThrusterTask requires "
```

Comment:
I'd expect an exception from the task state machine. 
```
gzmsg << "rock_gazebo::ThrusterTask requires joint names to be set. Going into exception" << endl;
exception(NO_JOINTS_NAMES);
```

Or would the task transit to exception state with the `throw`?

Reply:
It would have, but it would have been a plain EXCEPTION, which is indeed not what people usually expect. Changed it. I removed the message, though, since the state is explanatory enough I think.